### PR TITLE
Add column partitions in system.query_log to show which partitions are used for each MergeTree table

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add column partitions in system.query_log to show which partitions are used for each MergeTree table

### CODE REVIEW

File: src/Interpreters/QueryLog.cpp [hunk 1]
```diff
@@ -64,16 +64,15 @@ NamesAndTypesList QueryLogElement::getNamesAndTypes()
...
-        {"views", std::make_shared<DataTypeArray>(
-            std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"databases", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"tables", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"columns", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"projections", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"views", std::make_shared<DataTypeArray>(std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()))},
+        {"partitions",
+         std::make_shared<DataTypeMap>(
+             std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()),
```
Comment (Algunenano) [inline src/Interpreters/QueryLog.cpp:74]: 2 open questions (from complete ignorance):
* is `Map(LowCardinality(String), Array(LowCardinality(String))` actually better than not using LC at all? I see that it was used for other types too, but it's unclear to me what  the benefit of using LC inside nested types is.
* Is there a way (maybe impossible, maybe too expensive to be hidden behind an option) to get parts too?

Comment (taiyang-li) [reply src/Interpreters/QueryLog.cpp:74]: I will add a perf test about the first question.

Recording parts is too heavy for system.query_log. In our production environment, a single query may cover above 10k parts which will slow down query execution.

**Comment by kssenii** [comment]
@taiyang-li, are you going to continue with this PR?

**Comment by taiyang-li** [comment]
> @taiyang-li, are you going to continue with this PR?

I'm still working on it.  It will be set 'ready for review' when I finish.

**Comment by taiyang-li** [comment]
Already implemented in another pr. close this pr now.

**Pull request closed without merging**